update canary upgrade doc for in-cluster operator. (#8589)
istio/istio.io: content/en/docs/setup/install/operator/index.md

  --set tag={{< istio_full_version >}} \
  --set operatorNamespace=istio-operator \
  --set watchedNamespaces=istio-system \
  --set revision=1-8-1
{{< /text >}}

Note that you need to [download the Istio release](/docs/setup/getting-started/#download)
to run the above command.
{{< /tip >}}

> Copy the `IstioOperator` CR named `example-istiocontrolplane` from the cluster as `example-istiocontrolplane-1-8-1.yaml`, update the `IstioOperator` CR name to `example-istiocontrolplane-1-8-1` and add `revision: 1-8-1` to `IstioOperator` CR, the updated `IstioOperator` CR should resemble the following snippet:

Reviewer: ```suggestion
Make a copy of the `example-istiocontrolplane` CR and save it in a file named `example-istiocontrolplane-1-8-1.yaml`.
Change the name to `example-istiocontrolplane-1-8-1` and add `revision: 1-8-1` to the CR.
Your updated `IstioOperator` CR should look something like this:
```

Author: done, thanks!

{{< text bash >}}
$ cat example-istiocontrolplane-1-8-1.yaml
apiVersion: install.istio.io/v1alpha1
kind: IstioOperator
metadata:
  namespace: istio-system
  name: example-istiocontrolplane-1-8-1
spec:

>   revision: 1-8-1

Reviewer 2: Should also add `profile: default`?

Author: done

Reviewer 3: default is the default if profile is unset - we should be consistent with what we have elsewhere, which is to omit it, otherwise it may confuse users to mix these two.

  profile: default
{{< /text >}}

> Apply the updated `IstioOperator` CR to the cluster, then you will have two control plane deployments and services running side-by-side:

Reviewer: ```suggestion
Apply the updated `IstioOperator` CR to the cluster. After that, you will have two control plane deployments and services running side-by-side:
```

Author: done

istiod-1-8-1-597475f4f6-bgtcz   1/1     Running   0          64s
istiod-6ffcc65b96-bxzv5         1/1     Running   0          2m11s
{{< /text >}}

{{< text bash >}}
$ kubectl get svc -n istio-system -l app=istiod
NAME           TYPE        CLUSTER-IP       EXTERNAL-IP   PORT(S)                                         AGE
istiod         ClusterIP   10.104.129.150   <none>        15010/TCP,15012/TCP,443/TCP,15014/TCP,853/TCP   2m35s
istiod-1-8-1   ClusterIP   10.111.17.49     <none>        15010/TCP,15012/TCP,443/TCP,15014/TCP           88s
{{< /text >}}

To complete the upgrade, label the workload namespaces with `istio.io/rev=1-8-1` and restart the workloads, as
explained in the [Data plane upgrade](/docs/setup/upgrade/canary/#data-plane) documentation.

## Uninstall

If you used the operator to perform a canary upgrade of the control plane, you can uninstall the old control plane and keep the new one by deleting the old in-cluster `IstioOperator` CR, which will uninstall the old revision of Istio:

{{< text bash >}}
$ kubectl delete istiooperators.install.istio.io -n istio-system example-istiocontrolplane
{{< /text >}}

Wait until Istio is uninstalled - this may take some time.

> Delete the specific revision of Istio operator by running the following command:

Reviewer: This instruction isn't clear. If we had not deleted the old CR in the previous step, wouldn't the following command also remove it? If so, why would we run both commands, instead of just the second command? What use case would only do the first command and not the second? I think this should be explained more clearly.

Author: Emm, deleting the operator `istioctl operator remove`  before `IstioOperator` CR is fully removed will result in leftover Istio resources.

So we have to run  both commands and in correct order.

I updated the uninstall steps, PTAL

{{< text bash >}}

> $ istioctl operator remove --revision <revision>

Reviewer 2: As we are using `1-8-1` as an example of revision. I think following will be better?
```
$ istioctl operator remove --revision 1-8-1
```

Author: Normally, new revision of istio operator should not removed, we just tell the end users how to remove unused revision of istio operator(old one in upgrade, new one in rollback)

  --set tag={{< istio_full_version >}} \
  --set operatorNamespace=istio-operator \
  --set watchedNamespaces=istio-system \
  --set revision=1-8-1
{{< /text >}}

Note that you need to [download the Istio release](/docs/setup/getting-started/#download)
to run the above command.
{{< /tip >}}

Make a copy of the `example-istiocontrolplane` CR and save it in a file named `example-istiocontrolplane-1-8-1.yaml`.
Change the name to `example-istiocontrolplane-1-8-1` and add `revision: 1-8-1` to the CR.
Your updated `IstioOperator` CR should look something like this:

{{< text bash >}}
$ cat example-istiocontrolplane-1-8-1.yaml
apiVersion: install.istio.io/v1alpha1
kind: IstioOperator
metadata:
  namespace: istio-system
  name: example-istiocontrolplane-1-8-1
spec:
  revision: 1-8-1

>   profile: default

Reviewer: Please remove as suggested by @ostromart 
```suggestion
```

istiod-1-8-1-597475f4f6-bgtcz   1/1     Running   0          64s
istiod-6ffcc65b96-bxzv5         1/1     Running   0          2m11s
{{< /text >}}

{{< text bash >}}
$ kubectl get svc -n istio-system -l app=istiod
NAME           TYPE        CLUSTER-IP       EXTERNAL-IP   PORT(S)                                         AGE
istiod         ClusterIP   10.104.129.150   <none>        15010/TCP,15012/TCP,443/TCP,15014/TCP,853/TCP   2m35s
istiod-1-8-1   ClusterIP   10.111.17.49     <none>        15010/TCP,15012/TCP,443/TCP,15014/TCP           88s
{{< /text >}}

To complete the upgrade, label the workload namespaces with `istio.io/rev=1-8-1` and restart the workloads, as
explained in the [Data plane upgrade](/docs/setup/upgrade/canary/#data-plane) documentation.

## Uninstall

If you used the operator to perform a canary upgrade of the control plane, you can uninstall the old control plane and keep the new one by deleting the old in-cluster `IstioOperator` CR, which will uninstall the old revision of Istio:

{{< text bash >}}
$ kubectl delete istiooperators.install.istio.io -n istio-system example-istiocontrolplane
{{< /text >}}

Wait until Istio is uninstalled - this may take some time.

> Then you can remove the old of Istio operator with its revision by running the following command:

Reviewer: ```suggestion
Then you can remove the Istio operator for the old revision by running the following command:
```

{{< text bash >}}
$ istioctl operator remove --revision <revision>
{{< /text >}}

If you omit the `revision` flag, then all revisions of Istio operator will be removed.

> Note that deleting the operator before Istio or `IstioOperator` CR is fully removed may result in leftover Istio resources.

Reviewer: ```suggestion
Note that deleting the operator before the `IstioOperator` CR and corresponding Istio revision are fully removed may result in leftover Istio resources.
```
